src/pulses.jl

```diff
@@ -84,23 +84,39 @@ end
 Id(c::Channel) = Id(c, c.shape_params[:length])
 
 
-function AC(q::Qubit, num)
-
-	pulses = [
-		q -> Id(q),
-		q -> X90(q),
-		q -> X(q),
-		q -> X90m(q),
-		q -> Y90(q),
-		q -> Y(q),
-		q -> Y90m(q),
-		q -> Z90(q),
-		q -> Z(q),
-		q -> Z90m(q)
-	]
-
-	return pulses[num](q)
-
+function AC(q::Qubit, num; sampling_rate = 1.2e9)
+	if 0<num<=12
```

Comment:
You are using `num` to index into a vector, but `num` can be 0.  Did you intent to have 0 be a zero length identity?